src/services/auth.js

```diff
@@ -48,6 +49,20 @@ passport.use(new HTTPSignatureStrategy(
       })
   }))
 
+passport.use(new ClientCertStrategy((userName, info, done) => {
+  if (!config.auth.client_certificates_enabled) {
+    return done(new UnauthorizedError('Unsupported authentication method'))
+  }
+
+  Account.findByName(userName)
+    .then(function (userObj) {
+      if (!userObj || userObj.is_disabled || userObj.fingerprint !== info.fingerprint) {
```

Comment:
> Would this mean that other authentication methods would be removed?

I think we could only use one authentication method at a time. Is there a reason to use multiple?

> Do you mean putting the username in the common name field? 

Yes, but we may instead use the fingerprint as the username and the common name won't be used for authentication.